app/app-upgrade-client/src/components/nav/nav.tsx

```diff
@@ -0,0 +1,240 @@
+import {
+  AppBar,
+  Button,
+  Divider,
+  Drawer,
+  IconButton,
+  Link,
+  List,
+  ListItem,
+  ListItemText,
+  Toolbar,
+  Typography
+} from '@material-ui/core';
+import { makeStyles } from '@material-ui/core/styles';
+import CloseIcon from '@material-ui/icons/Close';
+import MenuIcon from '@material-ui/icons/Menu';
+import React, { FC } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { useLocation } from 'react-router-dom';
+import { signOutUser } from '../../auth/auth.slice';
+import logo from '../../images/appUpgrade_white.png';
+import { RootState } from '../../root-reducer';
+import { appLinks } from './appLinks';
+
+const drawerWidth = 240;
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    flexGrow: 1
+  },
+  title: {
+    flexGrow: 1,
+    position: 'relative',
+    '&:hover': {
+      cursor: 'pointer'
+    }
+  },
+  navLogo: {
+    height: 40,
+    width: 'auto'
+  },
+  menuLinks: {
+    [theme.breakpoints.down('sm')]: {
+      display: 'none'
+    }
+  },
+  menuButton: {
+    [theme.breakpoints.up('md')]: {
+      display: 'none'
+    }
+  },
+  menuButtonLink: {
+    marginRight: theme.spacing(2),
+    display: 'inline-block',
+    fontSize: '18px',
+    color: theme.palette.primary.contrastText,
+    '&:hover': {
+      cursor: 'pointer',
+      color: '#F5F5F5'
+    }
+  },
+  menuButtonUser: {
+    marginLeft: theme.spacing(4)
+  },
+  closeMenuButton: {
+    marginRight: 'auto',
+    marginLeft: 0
+  },
+  drawerPaper: {
+    width: drawerWidth,
+    [theme.breakpoints.up('md')]: {
+      display: 'none'
+    }
+  },
+  logoutLoginDrawerWrapper: {
+    padding: '8px 16px',
+
+    '& button': {
+      marginLeft: '0px !important'
+    }
+  }
+}));
+
+export const Nav: FC = () => {
+  const { pathname } = useLocation();
+  // const pathname = location.pathname;
+  const { user } = useSelector((state: RootState) => {
+    return {
+      user: state.auth.user
+    };
+  });
+  const [mobileMenuOpen, setMobileMenuOpen] = React.useState(false);
+  const classes = useStyles();
+  const dispatch = useDispatch();
+
+  function toggleMobileMenu() {
```

Comment:
For consistency we should do

const toggleMobileMenu = () => ...

Reply:
done